README.md

```diff
@@ -45,15 +45,18 @@ As go plugins can be finicky to correctly compile and install, you may want to c
 # We use go modules for everything.
 > export GO111MODULE=on
 
+# (Optionally) clean go cache
+go clean -modcache; go clean -cache
+
```

Comment:
```suggestion
```
Any reason why this needs to be there ?

Reply:
While testing I noted that GO didn't reliable update go-ds-s3 source. Clearing cache helped out. 
I also just noted that cleaning go cache has the nice side effect that building succeeded on [first try of building](https://github.com/ipfs/go-ds-s3/blob/2c56540bb971423f7f39d46839532f47278fc86d/README.md?plain=1#L62).

Anyway, it would also be fine for me to remove these lines. But i guess they can be useful aswell. What are your thoughts on this?

Reply:
Sounds aggressive to remove everything cached. But it's marked optional.
You don't need `go clean -cache` tho, the build cache isn't used for versioning caching.
Also in case you want to clean multiple things you can do `go clean -modcache -cache btw.

```suggestion
# (Optionally) clean go cache
go clean -modcache
```